protocols/v2/codec-sv2/src/error.rs

```diff
@@ -1,69 +1,105 @@
+//! # Error Handling and Result Types
+//!
+//! This module defines error types and utilities for handling errors in the `codec_sv2` module.
+//! It includes the `Error` enum for representing various errors, a C-compatible `CError` enum for
+//! FFI, and a `Result` type alias for convenience.
+
 #[cfg(test)]
 use core::cmp;
 use core::fmt;
 use framing_sv2::Error as FramingError;
 #[cfg(feature = "noise_sv2")]
 use noise_sv2::{AeadError, Error as NoiseError};
 
+/// A type alias for results returned by the `codec_sv2` modules.
+///
+/// `Result` is a convenient wrapper around the `core::result::Result` type,
+/// using the `Error` enum defined in this crate as the error type.
 pub type Result<T> = core::result::Result<T, Error>;
 
+/// Enumeration of possible errors in the `codec_sv2` module.
+///
+/// This enum represents various errors that can occur within the `codec_sv2` module, including
```

Comment:
I think using `codec_sv2 crate` instead of `.. module` would be more consistent.

Also, you can make the crate names clickable in docs.rs by doing the following:
```md
/// Enumeration of possible errors in the [`codec_sv2`] module.
///
/// This enum represents various errors that can occur within the [`codec_sv2`] module, including
/// errors from related crates like [`binary_sv2`], [`framing_sv2`], and [`noise_sv2`].
///
/// [`codec_sv2`]: https://docs.rs/codec_sv2
/// [`binary_sv2`]: https://docs.rs/binary_sv2
/// [`framing_sv2`]: https://docs.rs/framing_sv2
/// [`noise_sv2`]: https://docs.rs/noise_sv2
```